### PR TITLE
[CFM-1305] Add rearrange functionality to Help pages

### PR DESCRIPTION
**Pull request by ybaras**
[CFM-1305](https://issuetracker.amplexor.com/jira/browse/CFM-1305)

### CODE REVIEW

**Comment by ybaras** [comment]
@ordavidil Ready for review (sniffer fails on files unrelated to this PR).

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module [hunk 1]
```diff
@@ -6,3 +6,39 @@
...
+}
+
+/**
+ * Page callback for the help pages rearrange menu item.
+ *
+ * @return mixed
+ *   Returns help and guidance rearrange view.
+ */
+function c4m_content_book_help_rearrange_page() {
+  drupal_set_title('Rearrange');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.module]: Why not being explicit with __Rearrange help pages__ (or something similar) ?

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.views.inc [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @file
+ * Views plugins.
+ */
+
+/**
+ * Implements hook_views_plugins().
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.views.inc]: Fix the docs to explain what's the usage of this plugin.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/includes/c4m_content_book_plugin_argument_default.inc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * @file
+ * Views argument plugin.
+ */
+
+// @codingStandardsIgnoreFile
+
+/**
+ * The help pages argument default handler.
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/includes/c4m_content_book_plugin_argument_default.inc]: Elaborate why do we need this handler.

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.views_default.inc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+/**
+ * Implements hook_views_default_views().
+ */
+function c4m_content_book_views_default_views() {
+  $export = array();
+
+  $view = new view();
+  $view->name = 'c4m_help_guidance';
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/c4m_content_book.views_default.inc]: Shouldn't it include "rearrange" somewhere ?

File: project/profiles/capacity4more/modules/c4m/content/c4m_content_book/includes/c4m_content_book_plugin_argument_default.inc [hunk 5]
```diff
@@ -19,21 +19,33 @@ class c4m_content_book_plugin_argument_default extends views_plugin_argument_def
...
       ->range(0, 1)
       ->execute();
-    $entity = array_keys($entities['node']);
+
+    // If the Help & Guidance book ID could not be determined, add a human
+    // friendly error message and return early -which will cause Drupal to
+    // display `The requested page could not be found`.
+    if (!isset($result['node'])) {
+      drupal_set_message(t('Unable to retrieve help pages. Please <a href="@contact-page">contact the site administrator</a>.', array(
+        '@contact-page' => url('contact', array('purl' => array('disabled' => TRUE))),
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/includes/c4m_content_book_plugin_argument_default.inc]: @ybaras 
take the whole link out of the `t()` function and use a variable.
you can use `l()` to create the anchor element

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/content/c4m_content_book/includes/c4m_content_book_plugin_argument_default.inc]:
```php
l(t('contact the site administrator'), 'contact', $options);
```

**Comment by ybaras** [comment]
cfm-1305

**Comment by ordavidil** [comment]
Added a settings page to set the main book
selection_300

**Pull request merged**